### PR TITLE
Reintroduce OpenCensus trace bridge

### PR DESCRIPTION
**Pull request by dashpole**
Part of https://github.com/open-telemetry/opentelemetry-go/issues/2808.

This restores only the trace portion of the opencensus bridge without modification.  The metric portion needs to be rewritten, and will be done in a follow-up.

### CODE REVIEW

**Review by MrAlias — APPROVED** [review]

**Review by MadVikingGod — APPROVED** [review]

File: bridge/opencensus/internal/oc2otel/span_context.go [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+package oc2otel // import "go.opentelemetry.io/otel/bridge/opencensus/internal/oc2otel"
+
+import (
+	octrace "go.opencensus.io/trace"
+
+	"go.opentelemetry.io/otel/trace"
+)
+
+func SpanContext(sc octrace.SpanContext) trace.SpanContext {
```
Comment (hanyuancheung) [inline bridge/opencensus/internal/oc2otel/span_context.go:23]:
```suggestion
// SpanContext returns from OpenCensus SpanContext to an OpenTelemetry trace SpanContext.
func SpanContext(sc octrace.SpanContext) trace.SpanContext {
```

File: bridge/opencensus/internal/oc2otel/attributes.go [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+package oc2otel // import "go.opentelemetry.io/otel/bridge/opencensus/internal/oc2otel"
+
+import (
+	octrace "go.opencensus.io/trace"
+
+	"go.opentelemetry.io/otel/attribute"
+)
+
+func Attributes(attr []octrace.Attribute) []attribute.KeyValue {
```
Comment (hanyuancheung) [inline bridge/opencensus/internal/oc2otel/attributes.go:23]:
```suggestion
// Attributes converts from OpenCensus attribute keys and values to an OpenTelemetry attribute kvs.
func Attributes(attr []octrace.Attribute) []attribute.KeyValue {
```

File: bridge/opencensus/internal/oc2otel/tracer_start_options.go [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+import (
+	"fmt"
+
+	octrace "go.opencensus.io/trace"
+
+	"go.opentelemetry.io/otel/trace"
+)
+
+func StartOptions(optFuncs []octrace.StartOption) ([]trace.SpanStartOption, error) {
```
Comment (hanyuancheung) [inline bridge/opencensus/internal/oc2otel/tracer_start_options.go:25]:
```suggestion
// StartOptions returns from OpenCensus StartOption to an OpenTelemetry SnapStartOption.
func StartOptions(optFuncs []octrace.StartOption) ([]trace.SpanStartOption, error) {
```

**Comment by dashpole** [comment]
If we want to make changes to the trace bridge, i'd prefer to make those on the main branch, rather than on the new_sdk/main branch.

**Comment by MrAlias** [comment]
> If we want to make changes to the trace bridge, i'd prefer to make those on the main branch, rather than on the new_sdk/main branch.

Agreed. @hanyuancheung please add a PR to include your suggestions to the main branch. I'm going to merge this without them.

**Pull request merged**